lib/bot_saves_princess.rb

```diff
@@ -21,3 +22,10 @@ def display_path_to_princess(n,grid)
    end
   end
 end
+
+def validate_size_input(n)
+  #
+  if !n.odd? || n < 3 || n >= 100
+    puts 'RE-ENTER INPUT PARAMETERS'
+  end
+end
```

Comment:
This method validates that size input (n) is odd, larger than 3, and less than 100